runtime/Makefile

```diff
@@ -51,7 +51,7 @@ $(BT_OBJS): %.o: %.c
 	$(CLANG) -DHAVE_CONFIG_H -funwind-tables -frandom-seed=$< $(BT_WARN_FLAGS) $(CFLAGS) -c -o $@ $<
 
 $(DTOA_OBJS) $(DN_OBJS): %.o: %.c
-	$(CLANG) $(DEFINES) $(CFLAGS) -c -o $@ $<
+	$(CLANG) $(DEFINES) -O1 -c -o $@ $<
```

Comment:
@jclark there seems to be a bug in `decNumber` library with address space randomization. When compiled with `-02`  (and I presume there is some operating system support needed as well since this issue happens only in Ubuntu, and [disabling it OS level](https://gcc.gnu.org/wiki/Randomization) seems to fix it). We get a segfault for [`12-decimal/mul7-v.bal`](https://github.com/ballerina-platform/nballerina/blob/main/compiler/testSuite/12-decimal/mul7-v.bal). This is the relevant part of running Valgrind on it which is complaining about functions in `DecNumber` library
```
Memcheck, a memory error detector
Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
Command: ./mul7-v.exe

Use of uninitialised value of size 8
   at 0x10FAA8: decQuadFromBCD (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x1109D7: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x117B5D: decQuadMultiply (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10EB8A: _bal_decimal_mul (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10A62E: _B04rootmain (mul7-v.bal:21)
   by 0x10C7FA: main (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
 Uninitialised value was created by a stack allocation
   at 0x117A80: decQuadMultiply (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)

Use of uninitialised value of size 8
   at 0x10FACD: decQuadFromBCD (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x1109D7: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x117B5D: decQuadMultiply (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10EB8A: _bal_decimal_mul (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10A62E: _B04rootmain (mul7-v.bal:21)
   by 0x10C7FA: main (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
 Uninitialised value was created by a stack allocation
   at 0x1105F0: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)

Use of uninitialised value of size 8
   at 0x10FAFB: decQuadFromBCD (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x1109D7: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x117B5D: decQuadMultiply (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10EB8A: _bal_decimal_mul (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10A62E: _B04rootmain (mul7-v.bal:21)
   by 0x10C7FA: main (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
 Uninitialised value was created by a stack allocation
   at 0x1105F0: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)

Use of uninitialised value of size 8
   at 0x10FB20: decQuadFromBCD (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x1109D7: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x117B5D: decQuadMultiply (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10EB8A: _bal_decimal_mul (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10A62E: _B04rootmain (mul7-v.bal:21)
   by 0x10C7FA: main (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
 Uninitialised value was created by a stack allocation
   at 0x1105F0: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)

Invalid read of size 2
   at 0x10FB20: decQuadFromBCD (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x1109D7: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x117B5D: decQuadMultiply (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10EB8A: _bal_decimal_mul (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10A62E: _B04rootmain (mul7-v.bal:21)
   by 0x10C7FA: main (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
 Address 0x14ab6e is not stack'd, malloc'd or (recently) free'd


Process terminating with default action of signal 11 (SIGSEGV)
 Access not within mapped region at address 0x14AB6E
   at 0x10FB20: decQuadFromBCD (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x1109D7: decFinalize (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x117B5D: decQuadMultiply (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10EB8A: _bal_decimal_mul (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
   by 0x10A62E: _B04rootmain (mul7-v.bal:21)
   by 0x10C7FA: main (in /home/heshan/Projects/nballerina/test/out/12-decimal/result/mul7-v.exe)
 If you believe this happened as a result of a stack
 overflow in your program's main thread (unlikely but
 possible), you can try to increase the size of the
 main thread stack using the --main-stacksize= flag.
 The main thread stack size used in this run was 10000003072.
```

Reply:
Fixed